core/src/test/scala/unit/kafka/server/MetadataCacheTest.scala

```diff
@@ -167,70 +86,60 @@ class MetadataCacheTest {
     val topic0 = "topic-0"
     val topic1 = "topic-1"
 
-    val zkVersion = 3
-    val controllerId = 2
-    val controllerEpoch = 1
+    val topicIds = new util.HashMap[String, Uuid]()
+    topicIds.put(topic0, Uuid.randomUuid())
+    topicIds.put(topic1, Uuid.randomUuid())
 
-    def endpoints(brokerId: Int): Seq[UpdateMetadataEndpoint] = {
+    def endpoints(brokerId: Int): BrokerEndpointCollection = {
       val host = s"foo-$brokerId"
-      Seq(
-        new UpdateMetadataEndpoint()
+      new BrokerEndpointCollection(Seq(
+        new BrokerEndpoint()
           .setHost(host)
           .setPort(9092)
           .setSecurityProtocol(SecurityProtocol.PLAINTEXT.id)
-          .setListener(ListenerName.forSecurityProtocol(SecurityProtocol.PLAINTEXT).value),
-        new UpdateMetadataEndpoint()
+          .setName(ListenerName.forSecurityProtocol(SecurityProtocol.PLAINTEXT).value),
+        new BrokerEndpoint()
           .setHost(host)
           .setPort(9093)
           .setSecurityProtocol(SecurityProtocol.SSL.id)
-          .setListener(ListenerName.forSecurityProtocol(SecurityProtocol.SSL).value)
-      )
+          .setName(ListenerName.forSecurityProtocol(SecurityProtocol.SSL).value)
+      ).iterator.asJava)
     }
 
     val brokers = (0 to 4).map { brokerId =>
-      new UpdateMetadataBroker()
-        .setId(brokerId)
-        .setEndpoints(endpoints(brokerId).asJava)
+      new RegisterBrokerRecord()
+        .setBrokerId(brokerId)
+        .setEndPoints(endpoints(brokerId))
         .setRack("rack1")
     }
 
+    val topic0Record = new TopicRecord().setName(topic0).setTopicId(topicIds.get(topic0))
+    val topic1Record = new TopicRecord().setName(topic1).setTopicId(topicIds.get(topic1))
+    MetadataCacheTest.updateCache(cache, brokers ++ Seq(topic0Record, topic1Record))
```

Comment:
Is there a reason why we have this separate `updateCache` call instead of doing it all together like before?